File: Python/0062.py
class Solution(object):
   def uniquePaths(self, m, n):
      row = n
      column = m
      dp = [[-1 for i in range(m)] for j in range(n)]
      dp[row-2][column-1] = 1
      for i in range(column):
         dp[n-1][i] = 1
      for i in range(row):
         dp[i][column-1]=1
      for i in range(row-2,-1,-1):
         for j in range(column-2,-1,-1):
            dp[i][j] = dp[i+1][j] + dp[i][j+1]
      return dp[0][0]

File: Java/0062.java
class Solution {
    public int uniquePaths(int m, int n) {
        int[] arr = new int[m];
        arr[0] = 1;
        for(int i = 0; i < n; i++){
            for(int j = 1; j < m; j++){
                 arr[j] = arr[j-1] + arr[j];
            }
        }
    return arr[m-1];
    }
}
